text/0000-docs-information-architecture

```diff
@@ -0,0 +1,154 @@
+This design also proposes a possibly heuristic for determining what we can reasonably commit to maintaining as part of Gatsby core docs, and what might be too far outside the scope of core docs (either because the topics don't fit or because they are not a priority, given our limited time and resources). includes a list of what will *not* be included in Gatsby core docs.
+
+# Detailed design
+
+Here is a possible restructure of the current docs information architecture (with the inclusion of many docs we still need to write):
+
+Tutorials
+- Quick Start
+- Build and deploy a site
+    - Introduction
+    - 0. Setting up the development environment
+    - 1. Creating pages
+    - 2. Adding styles
+    - 3. Using layouts to build reusable site sections
+    - 4. Deploying a site live online (debating whether to keep this inside part 1, where it currently is)
+- Build a blog with GraphQL
+    - 5. Querying data for your blog
+    - 6. Source plugins and rendering queried data
+    - 7. Transformer plugins and transforming data into site content
+    - 8. Programmatically creating pages from data
+- Adding images to your site (or "Adding svg or blur-up effect to images)*
+- Creating your own plugin (tutorial for creating a plugin we've already vetted as easy to create)*
+- E-commerce*
+- User authentication*
+
+Reference docs
+- How Gatsby works
+    - The Gatsby core philosophy*
+    - Basic structure of a Gatsby site (includes config.js)*
+    - How Gatsby works with GraphQL (includes Node interface and how GraphQL works)*
+    - How Gatsby works with React (API specification)*
+    - How Gatsby works with Redux (bound action creators)*
+    - How Gatsby works with plugins
+    - PRPL pattern
+    - Lifecycle APIs
+- API references
+    - @jlengstorf this idea is from you, don't know how to fill this in...*
+    - API specification
+    - Gatsby Browser APIs
+    - Gatsby Node APIs
+    - Gatsby SSR APIs 
+
+- Gatsby CLI commands
+- Basic tasks
+    - Setting up dev environment   
+        - Gatsby on Windows
+    - Customizing site structure
+        - html.js
+        - webpack config
+        - Environment variables (right place for this??)
+    - Adding content 
+        - Images
+        - Markdown pages
+        - adding tags and categories to blogposts
+        - adding a 404 page
+    - Querying for data with GraphQL
+    - Debugging
+        - Debugging HTML builds
+        - Debugging Replace Renderer API
+    - Adding dynamic functionality
+        - Building apps with Gatsby
+        - E-commerce
+        - authentication
+        - dynamically rendered navigation*
+    - Deploying
+    - Hosting
+        - Path Prefix
+    - Performance
+        - Search Engine Optimization (SEO)
+        - Browser 
+        - Setting up proper HTTP caching
+- Migration Guides
+    - v0 to v1
+    - v1 to v2*
+    - v0 to v2 (good idea??)*
```

Comment:
I think we could skip the _v0 to v2_ guide. The benefit doesn't seem worth the work - anyone wanting to go from v0 to v2 could follow the other guides in sequence.